lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -182,6 +182,17 @@ function _checkBucketPolicyActions(requestType, actions, log) {
     return evaluators.isActionApplicable(mappedAction, actions, log);
 }
 
+function _checkBucketPolicyResources(request, resource, log) {
+    if (!request || (Array.isArray(resource) && resource.length === 0)) {
```

Comment:
Can skip the parenthesis, does not matter much though.
```suggestion
    if (!request || !Array.isArray(resource) || resource.length === 0)
```